cmd/admin-handlers.go

```diff
@@ -1779,3 +1784,71 @@ func (a adminAPIHandlers) TraceHandler(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 }
+
+// ServiceInfoHandler - GET /minio/admin/v1/serviceinfo?type={serviceType}
+// ----------
+// Get service information
+func (a adminAPIHandlers) ServiceInfoHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := newContext(r, w, "ServiceInfo")
+
+	objectAPI := validateAdminReq(ctx, w, r)
+	if objectAPI == nil {
+		return
+	}
+	var jsonBytes []byte
+	var err error
+
+	vars := mux.Vars(r)
+	switch serviceType := vars["serviceType"]; serviceType {
+	case "etcd":
+		errorString := ""
+		if globalEtcdClient == nil {
+			writeErrorResponseJSON(ctx, w, errorCodes.ToAPIErr(ErrETCDNotConfigured), r.URL)
+			return
+		}
+		etcdEndPoint := Environment.Get(EnvETCDEndpoints, "")
+		minioDomain := Environment.Get(EnvMinioDomain, "")
+		publicIPs := Environment.Get(EnvETCDPublicIps, "")
+		if etcdEndPoint != "" {
+			if err := checkStatus(etcdEndPoint); err != nil {
+				errorString = err.Error()
+			}
+
+		}
+
+		etcdInfo := madmin.ETCDInfo{
+			ETCDEndpoint: etcdEndPoint,
+			Domain:       minioDomain,
+			PublicIPs:    publicIPs,
+			Error:        errorString,
+		}
+		// Marshal API response
+		jsonBytes, err = json.Marshal(etcdInfo)
+		if err != nil {
+			writeErrorResponseJSON(ctx, w, toAdminAPIErr(ctx, err), r.URL)
+			return
+		}
+	}
+
+	// Reply with storage information (across nodes in a
+	// distributed setup) as json.
+	writeSuccessResponseJSON(w, jsonBytes)
+}
+
+//checkStatus - ping an endpoint , return err in case of no connection
+func checkStatus(endpoint string) error {
```

Comment:
We could rename the parameter to `endpointStr` (or) we could parse the endpoint string before calling checkStatus and let the parameter type be `xnet.Endpoint`

Reply:
whys is this needed ?

Reply:
IMO `endpoint` should be of type `net.Endpoint` rather than string as endpoint is not a string w.r.t our code structure. So i suggested this way.